### PR TITLE
fix ValidatingWebhookConfiguration not being deployable because it do…

### PR DESCRIPTION
**Pull request by grosser**
…es not have a namespace

longterm fix is to stop keeping a list an actually checking against the api :/

```
[15:49:08] JobExecution failed: Kubernetes error foo bar the server could not find the requested resource

```

@zendesk/compute

### CODE REVIEW

File: plugins/kubernetes/app/models/kubernetes/role_validator.rb [hunk 1]
```diff
@@ -6,7 +6,8 @@ class RoleValidator
     VALID_LABEL_VALUE = /\A[a-zA-Z0-9]([-a-zA-Z0-9.]*[a-zA-Z0-9])?\z/.freeze # also used in js ... cannot use /i
 
     NAMESPACELESS_KINDS = [
-      'APIService', 'ClusterRoleBinding', 'ClusterRole', 'CustomResourceDefinition', 'Namespace', 'PodSecurityPolicy'
+      'APIService', 'ClusterRoleBinding', 'ClusterRole', 'CustomResourceDefinition', 'Namespace', 'PodSecurityPolicy',
```
Comment (uthark) [inline plugins/kubernetes/app/models/kubernetes/role_validator.rb]:
```suggestion
     'ComponentStatus',
'Namespace',
'Node',
'PersistentVolume',
'MutatingWebhookConfiguration',
'ValidatingWebhookConfiguration',
'CustomResourceDefinition',
'APIService',
'TokenReview',
'SelfSubjectAccessReview',
'SelfSubjectRulesReview',
'SubjectAccessReview',
'CertificateSigningRequest',
'ENIConfig',
'PodSecurityPolicy',
'NodeMetrics',
'PodSecurityPolicy',
'ClusterRoleBinding',
'ClusterRole',
'PriorityClass',
'StorageClass',
'VolumeAttachment',
```

Comment (uthark) [reply plugins/kubernetes/app/models/kubernetes/role_validator.rb]: kubectl api-resources -o wide

**Review by uthark — DISMISSED** [review]
Let's update with other kinds too. Agree, that it should be dynamic in the future.

**Review by adammw — APPROVED** [review]
:+1: with oleg's changes. will eventually need to do this properly if we ever want to be compatible with cluster-scoped CRDs...

File: plugins/kubernetes/app/models/kubernetes/role_validator.rb (hunk 1 above)
Comment (adammw) [reply plugins/kubernetes/app/models/kubernetes/role_validator.rb]: I would leave out ENIConfig since that's a custom CRD, but the rest :+1:

Comment (uthark) [reply plugins/kubernetes/app/models/kubernetes/role_validator.rb]: We already work with nonstandard APIGroups (i.e. ValidatingWebhookConfiguration comes from admissionregistration.k8s.io), by the way.  ENIConfig comes from crd.k8s.amazonaws.com API Group.

Comment (uthark) [reply plugins/kubernetes/app/models/kubernetes/role_validator.rb]: Removed ENIConfig from suggestion.

File: plugins/kubernetes/app/models/kubernetes/role_validator.rb [hunk 2]
```diff
@@ -5,9 +5,33 @@ class RoleValidator
...
+      'CertificateSigningRequest',
+      'PodSecurityPolicy',
+      'NodeMetrics',
+      'PodSecurityPolicy',
+      'ClusterRoleBinding',
+      'ClusterRole',
+      'PriorityClass',
+      'StorageClass',
+      'VolumeAttachment',
+      'ValidatingWebhookConfiguration'
```
Comment (uthark) [inline plugins/kubernetes/app/models/kubernetes/role_validator.rb]: I see that you added it, but see line 15 too. Now it's duplicated. :)

**Comment by grosser** [comment]
https://github.com/zendesk/samson/pull/3394

**Pull request merged**